src/test/clojurescript/clara/test_salience.cljs

```diff
@@ -0,0 +1,115 @@
+  [Temperature]
+  =>
+  (swap! salience-rule-output conj 50))
+
+(defrule salience-rule3
+  {:salience 0}
+  [Temperature]
+  =>
+  (swap! salience-rule-output conj 0))
+
+(defrule salience-rule4
+  {:salience -50}
+  [Temperature]
+  =>
+  (swap! salience-rule-output conj -50))
+
+
+(deftest test-salience
+  (doseq [[sort-fn
+           group-fn
+           expected-order]
+
+          [[:default-sort :default-group :forward-order]
+           [:default-sort :salience-group :forward-order]
+           [:default-sort :neg-salience-group :backward-order]
+
+           [:numeric-greatest-sort :default-group :forward-order]
+           [:numeric-greatest-sort :salience-group :forward-order]
+           [:numeric-greatest-sort :neg-salience-group :backward-order]
+
+
+           [:boolean-greatest-sort :default-group :forward-order]
+           [:boolean-greatest-sort :salience-group :forward-order]
+           [:boolean-greatest-sort :neg-salience-group :backward-order]
+
+
+           [:numeric-least-sort :default-group :backward-order]
+           [:numeric-least-sort :salience-group :backward-order]
+           [:numeric-least-sort :neg-salience-group :forward-order]
+
+           [:boolean-least-sort :default-group :backward-order]
+           [:boolean-least-sort :salience-group :backward-order]
+           [:boolean-least-sort :neg-salience-group :forward-order]]]
+
+    (let [numeric-greatest-sort (fn [x y]
+                                  (cond
+                                    (= x y) 0
+                                    (> x y) -1
+                                    :else 1))
+
+          numeric-least-sort (fn [x y]
+                               (numeric-greatest-sort y x))
+
+          salience-group-fn (fn [production]
+                              (or (some-> production :props :salience)
+                                0))
+
+          neg-salience-group-fn (fn [p]
+                                  (- (salience-group-fn p)))]
+
+      (defsession test-salience-session 'clara.test-salience
+                     :cache false
+                     :activation-group-sort-fn (condp = sort-fn
+                                                 :default-sort nil
+                                                 :numeric-greatest-sort numeric-greatest-sort
+                                                 :numeric-least-sort numeric-least-sort
+                                                 :boolean-greatest-sort  >
+                                                 :boolean-least-sort <)
+                     :activation-group-fn (condp = group-fn
+                                            :default-group nil
+                                            :salience-group salience-group-fn
+                                            :neg-salience-group neg-salience-group-fn))
+
+      ;; Ensure the rule output reflects the salience-defined order.
+      ;; independently of the order of the rules.
+      (dotimes [n 10]
```

Comment:
Re: dotimes and the comment, both were copied from the CLJ test for salience in `test-rules.clj`. My impression was the author ran it 10 times be extra sure it worked. 😅 Will check to make sure I didn't miss something.